src/pages/IstioRuleDetails/IstioRuleInfo.tsx

```diff
@@ -1,21 +1,41 @@
 import * as React from 'react';
 import RuleId from '../../types/RuleId';
 import { ToastNotification, ToastNotificationList, Col, Row } from 'patternfly-react';
-import IstioRuleInfoDescription from './IstioRuleInfo/IstioRuleInfoDescription';
 import * as API from '../../services/Api';
 import { RuleAction } from '../../types/IstioRuleInfo';
-import IstioRuleInfoAction from './IstioRuleInfo/IstioRuleInfoAction';
+import IstioRuleDetailsDescription from './IstioRuleDestailsDescription';
+import AceEditor, { AceOptions } from 'react-ace';
+import 'brace/mode/yaml';
+import 'brace/theme/eclipse';
+import './IstioRuleInfo.css';
 
-type RuleInfoState = {
+const yaml = require('js-yaml');
+
+interface RuleInfoState {
   name: string;
   match: string;
   actions: RuleAction[];
   error: boolean;
   errorMessage: string;
+}
+
+interface RuleDetailsId extends RuleId {
+  search?: string;
+}
+
+interface ParsedSearch {
+  type?: string;
+  name?: string;
+}
+
```

Comment:
Other thing I wanna add to the style code is use Proptypes. But LGFM until we have guidelines

Reply:
Can you add more details here ?
If you show me an example I can try to add it and follow the style.

Reply:
Here is an example 

```js
import PropTypes from 'prop-types';

class MyComponent extends React.Component {
  render() {
    // This must be exactly one element or it will warn.
    const children = this.props.children;
    const name = this.props.name;
    return (
      <div>
        The name is { name }
        {children}
      </div>
    );
  }
}

MyComponent.propTypes = {
  name: PropTypes.string,
  children: PropTypes.element.isRequired
};

// Specifies the default values for props:
MyComponent.defaultProps = {
  name: 'Stranger'
};
```

Here is an example documenting the different validators provided:

```js
import PropTypes from 'prop-types';

MyComponent.propTypes = {
  // You can declare that a prop is a specific JS type. By default, these
  // are all optional.
  optionalArray: PropTypes.array,
  optionalBool: PropTypes.bool,
  optionalFunc: PropTypes.func,
  optionalNumber: PropTypes.number,
  optionalObject: PropTypes.object,
  optionalString: PropTypes.string,
  optionalSymbol: PropTypes.symbol,

  // Anything that can be rendered: numbers, strings, elements or an array
  // (or fragment) containing these types.
  optionalNode: PropTypes.node,

  // A React element.
  optionalElement: PropTypes.element,

  // You can also declare that a prop is an instance of a class. This uses
  // JS's instanceof operator.
  optionalMessage: PropTypes.instanceOf(Message),

  // You can ensure that your prop is limited to specific values by treating
  // it as an enum.
  optionalEnum: PropTypes.oneOf(['News', 'Photos']),

  // An object that could be one of many types
  optionalUnion: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.instanceOf(Message)
  ]),

  // An array of a certain type
  optionalArrayOf: PropTypes.arrayOf(PropTypes.number),

  // An object with property values of a certain type
  optionalObjectOf: PropTypes.objectOf(PropTypes.number),

  // An object taking on a particular shape
  optionalObjectWithShape: PropTypes.shape({
    color: PropTypes.string,
    fontSize: PropTypes.number
  }),

  // You can chain any of the above with `isRequired` to make sure a warning
  // is shown if the prop isn't provided.
  requiredFunc: PropTypes.func.isRequired,

  // A value of any data type
  requiredAny: PropTypes.any.isRequired,

  // You can also specify a custom validator. It should return an Error
  // object if the validation fails. Don't `console.warn` or throw, as this
  // won't work inside `oneOfType`.
  customProp: function(props, propName, componentName) {
    if (!/matchme/.test(props[propName])) {
      return new Error(
        'Invalid prop `' + propName + '` supplied to' +
        ' `' + componentName + '`. Validation failed.'
      );
    }
  },

  // You can also supply a custom validator to `arrayOf` and `objectOf`.
  // It should return an Error object if the validation fails. The validator
  // will be called for each key in the array or object. The first two
  // arguments of the validator are the array or object itself, and the
  // current item's key.
  customArrayProp: PropTypes.arrayOf(function(propValue, key, componentName, location, propFullName) {
    if (!/matchme/.test(propValue[key])) {
      return new Error(
        'Invalid prop `' + propFullName + '` supplied to' +
        ' `' + componentName + '`. Validation failed.'
      );
    }
  })
};
```

Reply:
Yes, it is not necessary, my point of view is that there is some views with proptypes and others with not so....or we use proptype in all files or we should remove it

Reply:
If I understand correctly Proptypes is something that is primarily meant for vanilla JS, although it works as well with typescript but there is some overlap then. Maybe for all that is covered by typescript we can stick with normal typing, and use Proptypes only when we want more advanced validation?